lib/tictactoe.rb

```diff
@@ -1,42 +1,78 @@
+require 'ai_basic'
+require 'ai_minimax'
 require 'game_board'
 require 'game_rules'
+require 'game_state'
 require 'player_manager'
-require 'game_loop'
-require 'game_screen'
-class TicTacToe
 
-  attr_accessor :board, :game_loop
+class TicTacToe
 
-  def initialize (game_io, player1, player2)
+  def initialize
+    @rules = GameRules.new
+    @board = GameBoard.new
+    @players = PlayerManager.new
+    @ai_easy = AiBasic.new(@rules)
+    @ai_hard = AiMinimax.new(@rules)
+  end
 
-    @io = game_io
-    @players = PlayerManager.new(@io, player1, player2)
-    @legend = ['0', '1', '2', '3', '4', '5', '6', '7', '8']
-    @legend_size = 3
+  def new_game
     @board = GameBoard.new
-    @rules = GameRules.new
-    @game_screen = GameScreen.new(@io, @board, @players, @rules, @legend, @legend_size)
-    @game_loop = GameLoop.new(@board, @game_screen, @rules, @players)
   end
 
-  def play_game
-    begin
-    @game_loop.game_loop
-    new_game
-    end while play_again?
+  def get_board
+    @board.spaces
+  end
+
+  def game_over?
+    @rules.game_over?(@board)
+  end
+
+  def tied_game?
+    @rules.tied?(@board)
+  end
+
+  def winner
+    @rules.winner(@board)
+  end
+
+  def play_human_move(move)
+    if valid_move?(move)
+      @board.play_move(@players.current_player.mark, move)
+      @players.switch_turns
+    end
+  end
+
+  def valid_move?(move)
+    @board.valid_move?(move)
+  end
+
+  def get_ai_player_move
+    @players.current_player.type.get_move(get_game_state) if is_current_player_ai?
```

Comment:
It should just return nil if the player is human. Let me make sure this won't have any unintended results else where.